ReactWindows/ReactNative/Common/AccessibilityHelper.cs

```diff
@@ -0,0 +1,292 @@
+        /// </summary>
+        /// <param name="parentUIElement">The element where <paramref name="uiElement"/> is added as child.</param>
+        /// <param name="uiElement">The element being added to the visual tree.</param>
+        public static void InitImportantForAccessibility(UIElement parentUIElement, UIElement uiElement)
+        {
+            DispatcherHelpers.RunOnDispatcher(() =>
+            {
+                // Post on UI thread because the method relies on parent/child
+                // relationship has already been established and sometimes calls may occur while the
+                // element is in process of being added as a child.
+                InitImportantForAccessibilityInternal(parentUIElement, uiElement);
+            });
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="parentUIElement"></param>
+        /// <param name="uiElement"></param>
+            private static void InitImportantForAccessibilityInternal(UIElement parentUIElement, UIElement uiElement)
+        {
+            var parentImportantForAccessibilityAttached = GetImportantForAccessibilityAttached(parentUIElement);
+            var parentUIElementAutomationPeer = FrameworkElementAutomationPeer.FromElement(parentUIElement);
+            if (parentImportantForAccessibilityAttached == ImportantForAccessibility.Yes ||
+                parentImportantForAccessibilityAttached == ImportantForAccessibility.NoHideDescendants ||
+                IsHiddenByAncestor(parentUIElementAutomationPeer))
+            {
+                // If an ancestor is "hiding" the element, set AccessibilityView to Raw.
+                var uiElementAutomationPeer = FrameworkElementAutomationPeer.FromElement(uiElement);
+                AutomationProperties.SetAccessibilityView(uiElement, AccessibilityView.Raw);
+                SetChildrenAccessibilityView(uiElementAutomationPeer, AccessibilityView.Raw);
+            }
+        }
+
+        /// <summary>
+        /// Sets the ImportantForAccessibility property for <paramref name="uiElement"/>.
+        /// It uses AutomationProperties.AccessibilityView to expose the element and its children
+        /// to narrator. ImportantForAccessibility value is stored as an attached property.
+        /// </summary>
+        /// <param name="uiElement">Element which ImportantForAccessibility property is set.</param>
+        /// <param name="importantForAccessibility">The new value of ImportantForAccessibility property.</param>
+        public static void SetImportantForAccessibility(UIElement uiElement, ImportantForAccessibility importantForAccessibility)
+        {
+            // Post on UI thread because the method relies on parent/child
+            // relationship has already been established and sometimes calls may occur while the
+            // element is in process of being added as a child.
+            DispatcherHelpers.RunOnDispatcher(() =>
```

Comment:
>DispatcherHelpers.RunOnDispatcher(() => [](start = 12, length = 39)

See my concern in `InitImportantForAccessibility`

Reply:
Same here.

---
In reply to: [165803829](https://github.com/Microsoft/react-native-windows/pull/1595#discussion_r165803829) [](ancestors = 165803829)

Reply:
I'll close this and we'll use the other comment to track this feedback.

---
In reply to: [165824147](https://github.com/Microsoft/react-native-windows/pull/1595#discussion_r165824147) [](ancestors = 165824147,165803829)